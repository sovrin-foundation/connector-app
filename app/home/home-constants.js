// @flow

export const FEEDBACK_TEST_ID = 'home-feedback-id-label'
export const SOVRINTOKEN_TEST_ID = 'home-sovrintoken-id-label'
export const SOVRINTOKEN_AMOUNT_TEST_ID = 'home-sovrintoken-amount'

export const HOW_IT_WORKS =
  'You now have a digital wallet! Want to see how it works?'
export const ON_COMPUTER = 'On a computer'
export const GO_TO_FABER = ', go to www.evernym.com/faber'
export const USE_TUTORIAL = 'and use this app to go through the tutorial.'
