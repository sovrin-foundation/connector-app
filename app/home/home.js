// @flow
import React, { PureComponent } from 'react'
import {
  Animated,
  StyleSheet,
  Platform,
  Dimensions,
  View,
  Text,
} from 'react-native'
import { connect } from 'react-redux'
import firebase from 'react-native-firebase'
import {
  Container,
  CustomView,
  Icon,
  UserAvatar,
  CustomText,
  CustomHeader,
  Loader,
} from '../components'

import { createStackNavigator } from 'react-navigation'
import Bubbles from './bubbles'
import {
  color,
  barStyleDark,
  OFFSET_3X,
  OFFSET_2X,
  isBiggerThanShortDevice,
  isIphoneX,
  isBiggerThanVeryShortDevice,
  whiteSmokeSecondary,
  responsiveHorizontalPadding,
} from '../common/styles'
import { primaryHeaderStyles } from '../components/layout/header-styles'
import { homeRoute, walletRoute } from '../common'
import { getConnections } from '../store/connections-store'
import type { Store } from '../store/type-store'
import type { HomeProps, HomeState } from './type-home'
import {
  FEEDBACK_TEST_ID,
  SOVRINTOKEN_TEST_ID,
  SOVRINTOKEN_AMOUNT_TEST_ID,
  HOW_IT_WORKS,
  ON_COMPUTER,
  GO_TO_FABER,
  USE_TUTORIAL,
} from './home-constants'
import { Apptentive } from 'apptentive-react-native'
import WalletBalance from '../wallet/wallet-balance'
import type { Connection } from '../store/type-connection-store'
import Banner from '../components/banner/banner'
import { NavigationActions } from 'react-navigation'
import { getUnseenMessages } from '../store/store-selector'
import { scale } from 'react-native-size-matters'
import { size } from './../components/icon'

const { width, height } = Dimensions.get('window')

export class DashboardScreen extends PureComponent<HomeProps, HomeState> {
  state = {
    scrollY: new Animated.Value(0),
  }

  static navigationOptions = ({ navigation }) => ({
    header: (
      <CustomHeader
        backgroundColor={whiteSmokeSecondary}
        outerContainerStyles={styles.headerOuterContainer}
        largeHeader
      >
        <WalletBalance
          render={balance => (
            <CustomView
              row
              center
              onPress={() => {
                const navigateAction = NavigationActions.navigate({
                  routeName: walletRoute,
                  key: walletRoute,
                })
                navigation.dispatch(navigateAction)
              }}
              testID={SOVRINTOKEN_AMOUNT_TEST_ID}
            >
              <Icon
                small
                testID={SOVRINTOKEN_TEST_ID}
                src={require('../images/sovrinTokenOrange.png')}
              />
              <CustomText
                h5
                demiBold
                center
                style={[
                  styles.floatTokenAmount,
                  {
                    fontSize: tokenAmountSize(balance ? balance.length : 0),
                  },
                ]}
                transparentBg
                testID={SOVRINTOKEN_AMOUNT_TEST_ID}
                formatNumber
              >
                {balance}
              </CustomText>
            </CustomView>
          )}
        />

        <CustomView horizontalSpace>
          <Icon
            medium
            onPress={() => Apptentive.presentMessageCenter()}
            testID={FEEDBACK_TEST_ID}
            src={require('../images/icon_feedback_grey.png')}
          />
        </CustomView>
      </CustomHeader>
    ),
  })

  componentDidUpdate(prevProps: HomeProps) {
    const noUnSeenMessages =
      Object.keys(prevProps.unSeenMessages).length &&
      !Object.keys(this.props.unSeenMessages).length

    if (noUnSeenMessages) {
      firebase.notifications().setBadge(0)
    }
  }

  render() {
    const bubblesHeight = this.state.scrollY.interpolate({
      inputRange: [0, 5],
      outputRange: [0, -5],
      extrapolate: 'clamp',
    })

    const { connections: { data, hydrated }, unSeenMessages } = this.props
    // type casting from Array<mixed> to any and then to what we need
    // because flow Array<mixed> can't be directly type casted as of now
    const connections: Connection[] = (getConnections(data): any)
    const connectionsCheck = connections && connections.length > 0

    return (
      <Container tertiary>
        <Container tertiary>
          <Banner navigation={this.props.navigation} />
          {connectionsCheck ? (
            <Bubbles
              navigation={this.props.navigation}
              height={bubblesHeight}
              connections={connections}
              unSeenMessages={unSeenMessages}
            />
          ) : (
            <Container center>
              <View style={styles.messageBox}>
                <CustomText demiBold primary h4 transparentBg>
                  {HOW_IT_WORKS}
                </CustomText>
                <Text
                  style={[
                    styles.messageBoxBodyText,
                    { paddingTop: 20, paddingBottom: 20 },
                  ]}
                >
                  <Text style={{ color: color.actions.font.seventh }}>
                    {ON_COMPUTER}
                  </Text>
                  <Text
                    style={{
                      color: color.bg.secondary.color,
                    }}
                  >
                    {GO_TO_FABER}
                  </Text>
                </Text>
                <CustomText
                  h4
                  style={[
                    { color: color.bg.secondary.color, paddingBottom: '40%' },
                  ]}
                  transparentBg
                >
                  {USE_TUTORIAL}
                </CustomText>
              </View>
            </Container>
          )}
          {!hydrated ? (
            <Container center>
              <Loader type="dark" delay={1000} />
            </Container>
          ) : null}
          <CustomView style={[styles.userAvatarContainer]}>
            <UserAvatar />
          </CustomView>
        </Container>
      </Container>
    )
  }
}

const mapStateToProps = (state: Store) => {
  // when ever there is change in claimOffer state and proof request state
  // getUnseenMessages selector will return updated data
  // this data is used to update red dots on connection bubbles.
  let unSeenMessages = getUnseenMessages(state)
  return {
    connections: state.connections,
    unSeenMessages,
  }
}
export const getHeight = (height: number) => {
  if (isIphoneX) {
    return height - 300
  }
  if (isBiggerThanVeryShortDevice && !isBiggerThanShortDevice) {
    return height - 230
  }
  if (isBiggerThanShortDevice) {
    return height - 250
  }
  return height - 224
}
export default createStackNavigator({
  [homeRoute]: {
    screen: connect(mapStateToProps)(DashboardScreen),
  },
})

export const tokenAmountSize = (tokenAmountLength: number): number => {
  // this resizing logic is different than wallet tabs header
  switch (true) {
    case tokenAmountLength < 16:
      return scale(26)
    case tokenAmountLength < 20:
      return scale(20)
    default:
      return scale(19)
  }
}

const styles = StyleSheet.create({
  userAvatarContainer: {
    backgroundColor: 'transparent',
    width: size.extraLarge, // width of user avathar icon extraLarge
    position: 'absolute',
    left: (width - size.extraLarge) / 2,
    top: getHeight(height),
  },
  floatTokenAmount: {
    color: color.actions.font.seventh,
    paddingHorizontal: 8,
  },
  headerOuterContainer: {
    paddingHorizontal: responsiveHorizontalPadding,
  },
  messageBox: {
    width: '75%',
  },
  messageBoxBodyText: {
    fontSize: 20,
  },
})
