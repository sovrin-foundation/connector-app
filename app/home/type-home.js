// @flow
import type { ReactNavigation } from '../common/type-common'
import type {
  ConnectionStore,
  Connection,
} from '../store/type-connection-store'

export type BubbleProps = {
  size: number,
  image: ?string,
  testID: string,
  senderName: string,
  senderDID: string,
  radius: number,
  identifier: string,
  disableTaps: boolean,
  disableTopView: () => void,
  allowInteractions: boolean,
  showBadge: boolean,
} & ReactNavigation

export type BubbleState = {
  failed: boolean,
}

export type BubblesProps = {
  connections: Array<Connection>,
  height: number,
  unSeenMessages: {
    [string]: [string],
  },
} & ReactNavigation

export type HomeProps = {
  connections: ConnectionStore,
  unSeenMessages: {
    [string]: [string],
  },
} & ReactNavigation

export type HomeState = {
  // Don't have react-native Animated API type definition
  scrollY: any,
}

export type ConnectionBubblesState = {
  disableTaps: boolean,
  interactionsDone: boolean,
}
