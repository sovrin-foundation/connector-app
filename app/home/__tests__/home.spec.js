// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { CLAIM_OFFER_STATUS } from '../../claim-offer/type-claim-offer'
import {
  claimOfferRoute,
  homeTabRoute,
  authenticationRoute,
} from '../../common'
import {
  MESSAGE_TYPE,
  PUSH_NOTIFICATION_SENT_CODE,
} from '../../api/api-constants'
import { DashboardScreen } from '../home'
import {
  getNavigation,
  getStore,
  myPairWiseConnectionDetails,
  vcxSerializedConnection,
} from '../../../__mocks__/static-data'

function props(claimOfferStatus, noConnections) {
  let connectionsData = {
    '3nj819kkjywdppuje79': {
      identifier: '3nj819kkjywdppuje79',
      name: 'Test Connection',
      senderDID: '70075yyojywdppuje79',
      senderEndpoint: '34.216.340.155:3000',
      size: 100,
      logoUrl: 'https://logourl.com/logo.png',
      vcxSerializedConnection,
      ...myPairWiseConnectionDetails,
    },
  }

  if (noConnections) {
    connectionsData = {}
  }

  return {
    connections: {
      data: connectionsData,
      hydrated: true,
    },
    navigation: getNavigation(),
    claimOfferStatus: claimOfferStatus || CLAIM_OFFER_STATUS.RECEIVED,
    route: {
      currentScreen: homeTabRoute,
    },
    pushNotification: {
      notification: null,
    },
    getUserInfo: jest.fn(),
    pushNotificationReceived: jest.fn(),
    authenticationRequestReceived: jest.fn(),
    unSeenMessages: {},
  }
}

describe('<DashboardScreen />', () => {
  const store = getStore()

  jest.useFakeTimers()
  it('should render Home and redirect user to claim offer modal', () => {
    const dashboardProps = props(false, true)
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DashboardScreen {...dashboardProps} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Home and show loader', () => {
    const dashboardProps = props(false, true)
    dashboardProps.connections.hydrated = false
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DashboardScreen {...dashboardProps} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Home and show introductory text', () => {
    const dashboardProps = props(false, false)
    dashboardProps.connections.data = {}
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DashboardScreen {...dashboardProps} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
