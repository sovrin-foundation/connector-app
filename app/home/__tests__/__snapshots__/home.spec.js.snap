// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardScreen /> should render Home and redirect user to claim offer modal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
    tertiary={true}
  >
    <View
      style={Object {}}
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="wallet-backup-banner-label-touchable"
          accessibilityTraits={undefined}
          accessible={false}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          horizontalSpace={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          row={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d0021b",
              "flexDirection": "row",
              "paddingHorizontal": 15,
              "paddingVertical": 8,
            }
          }
          testID="wallet-backup-banner-label-touchable"
          vCenter={true}
          verticalSpace={true}
        >
          <View
            clearBg={true}
            iconStyle={
              Array [
                Object {
                  "height": 25,
                  "width": 29,
                },
                Object {
                  "tintColor": "white",
                },
              ]
            }
            src={1}
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <View
              style={Object {}}
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Array [
                    Object {
                      "height": 50,
                      "width": 50,
                    },
                    null,
                    Object {
                      "height": 25,
                      "width": 29,
                    },
                    Object {
                      "tintColor": "white",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            horizontalSpace={true}
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              You have new data that is not backed up.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  null,
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              Create another backup so you don't lose anything.
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "width": "75%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              Object {
                "fontWeight": "600",
              },
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              Object {
                "color": "#333333",
              },
              null,
              null,
              null,
              null,
            ]
          }
        >
          You now have a digital wallet! Want to see how it works?
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "paddingBottom": 20,
                "paddingTop": 20,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#EB9B2D",
              }
            }
          >
            On a computer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#777777",
              }
            }
          >
            , go to www.evernym.com/faber
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#777777",
                "paddingBottom": "40%",
              },
            ]
          }
        >
          and use this app to go through the tutorial.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "left": 120,
          "position": "absolute",
          "top": 316,
          "width": 80,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="user-avatar-touchable"
        accessibilityTraits={undefined}
        accessible={false}
        clearBg={true}
        extraLarge={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        resizeMode="cover"
        round={true}
        src={
          Object {
            "uri": "/var/application/DocumentDir/user-avatar.jpeg",
          }
        }
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
        testID="user-avatar-touchable"
      >
        <View
          style={Object {}}
        >
          <Image
            accessibilityLabel="user-avatar"
            accessible={true}
            resizeMode="cover"
            source={
              Object {
                "uri": "/var/application/DocumentDir/user-avatar.jpeg",
              }
            }
            style={
              Array [
                Object {
                  "height": 80,
                  "width": 80,
                },
                Object {
                  "borderRadius": 40,
                },
              ]
            }
            testID="user-avatar"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<DashboardScreen /> should render Home and show introductory text 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
    tertiary={true}
  >
    <View
      style={Object {}}
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="wallet-backup-banner-label-touchable"
          accessibilityTraits={undefined}
          accessible={false}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          horizontalSpace={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          row={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d0021b",
              "flexDirection": "row",
              "paddingHorizontal": 15,
              "paddingVertical": 8,
            }
          }
          testID="wallet-backup-banner-label-touchable"
          vCenter={true}
          verticalSpace={true}
        >
          <View
            clearBg={true}
            iconStyle={
              Array [
                Object {
                  "height": 25,
                  "width": 29,
                },
                Object {
                  "tintColor": "white",
                },
              ]
            }
            src={1}
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <View
              style={Object {}}
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Array [
                    Object {
                      "height": 50,
                      "width": 50,
                    },
                    null,
                    Object {
                      "height": 25,
                      "width": 29,
                    },
                    Object {
                      "tintColor": "white",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            horizontalSpace={true}
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              You have new data that is not backed up.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  null,
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              Create another backup so you don't lose anything.
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "width": "75%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              Object {
                "fontWeight": "600",
              },
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              Object {
                "color": "#333333",
              },
              null,
              null,
              null,
              null,
            ]
          }
        >
          You now have a digital wallet! Want to see how it works?
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "paddingBottom": 20,
                "paddingTop": 20,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#EB9B2D",
              }
            }
          >
            On a computer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#777777",
              }
            }
          >
            , go to www.evernym.com/faber
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#777777",
                "paddingBottom": "40%",
              },
            ]
          }
        >
          and use this app to go through the tutorial.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "left": 120,
          "position": "absolute",
          "top": 316,
          "width": 80,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="user-avatar-touchable"
        accessibilityTraits={undefined}
        accessible={false}
        clearBg={true}
        extraLarge={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        resizeMode="cover"
        round={true}
        src={
          Object {
            "uri": "/var/application/DocumentDir/user-avatar.jpeg",
          }
        }
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
        testID="user-avatar-touchable"
      >
        <View
          style={Object {}}
        >
          <Image
            accessibilityLabel="user-avatar"
            accessible={true}
            resizeMode="cover"
            source={
              Object {
                "uri": "/var/application/DocumentDir/user-avatar.jpeg",
              }
            }
            style={
              Array [
                Object {
                  "height": 80,
                  "width": 80,
                },
                Object {
                  "borderRadius": 40,
                },
              ]
            }
            testID="user-avatar"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<DashboardScreen /> should render Home and show loader 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  tertiary={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
    tertiary={true}
  >
    <View
      style={Object {}}
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="wallet-backup-banner-label-touchable"
          accessibilityTraits={undefined}
          accessible={false}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          horizontalSpace={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          row={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d0021b",
              "flexDirection": "row",
              "paddingHorizontal": 15,
              "paddingVertical": 8,
            }
          }
          testID="wallet-backup-banner-label-touchable"
          vCenter={true}
          verticalSpace={true}
        >
          <View
            clearBg={true}
            iconStyle={
              Array [
                Object {
                  "height": 25,
                  "width": 29,
                },
                Object {
                  "tintColor": "white",
                },
              ]
            }
            src={1}
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <View
              style={Object {}}
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Array [
                    Object {
                      "height": 50,
                      "width": 50,
                    },
                    null,
                    Object {
                      "height": 25,
                      "width": 29,
                    },
                    Object {
                      "tintColor": "white",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            horizontalSpace={true}
            style={
              Object {
                "paddingHorizontal": 15,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              You have new data that is not backed up.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 15,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#333333",
                    "color": "#ffffff",
                  },
                  null,
                  null,
                  Object {
                    "backgroundColor": "transparent",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                ]
              }
            >
              Create another backup so you don't lose anything.
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "width": "75%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              Object {
                "fontWeight": "600",
              },
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              Object {
                "color": "#333333",
              },
              null,
              null,
              null,
              null,
            ]
          }
        >
          You now have a digital wallet! Want to see how it works?
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "paddingBottom": 20,
                "paddingTop": 20,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#EB9B2D",
              }
            }
          >
            On a computer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#777777",
              }
            }
          >
            , go to www.evernym.com/faber
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#777777",
                "paddingBottom": "40%",
              },
            ]
          }
        >
          and use this app to go through the tutorial.
        </Text>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "left": 120,
          "position": "absolute",
          "top": 316,
          "width": 80,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel="user-avatar-touchable"
        accessibilityTraits={undefined}
        accessible={false}
        clearBg={true}
        extraLarge={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        resizeMode="cover"
        round={true}
        src={
          Object {
            "uri": "/var/application/DocumentDir/user-avatar.jpeg",
          }
        }
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
        testID="user-avatar-touchable"
      >
        <View
          style={Object {}}
        >
          <Image
            accessibilityLabel="user-avatar"
            accessible={true}
            resizeMode="cover"
            source={
              Object {
                "uri": "/var/application/DocumentDir/user-avatar.jpeg",
              }
            }
            style={
              Array [
                Object {
                  "height": 80,
                  "width": 80,
                },
                Object {
                  "borderRadius": 40,
                },
              ]
            }
            testID="user-avatar"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
